vignettes/base_R.Rmd

```diff
@@ -0,0 +1,235 @@
+---
+title: "Translating between `purrr` and base R"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Translating between `purrr` and base R}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.width = 7,
+  fig.height = 4.5,
+  fig.align = "center"
+)
+```
+
+```{r setup}
+library(purrr)
+```
+
+# Introduction
+Many functions in R are vectorized, such as `as.character` and `exp`, efficiently applying the operation of interest to each element of a vector without the need to write a loop. However, some functions do not have vectorized methods for some types of vectors, especially lists (rather than atomic vectors), and sometimes the function to be applied is not vectorized in an argument of interest. In base R, the built-in `*apply` family of functions apply the function of interest to each element of a vector or list, returning the results as a list, vector, or matrix. This simplifies the code compared to writing a loop, as the need to pre-define an empty container for the results and filling that container is obliterated. 
```

Comment:
Our style is to put parentheses after function names, like `as.character()`.

Reply:
OK